.github/workflows/build.yml

```diff
@@ -122,6 +117,14 @@ jobs:
           LDAP_PASSWORDS: "password1,password2,password3"
     steps:
       - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - name: Install setup-terraform dependencies
+        run: apt-get -y update && apt-get install unzip
+      - uses: hashicorp/setup-terraform@v2
```

Comment:
How does setting up terraform here prevent the intermittent CI failures? Is this an alternative to installing terraform (elsewhere), or a fallback?

Reply:
Basically, the terraform-plugin-sdk will look for a terraform binary in the system's path, if it doesn't find it then it will attempt to download it.

Reply:
And it downloads it for each test

Reply:
I'll let JM correct my explanation if I'm wrong since he explained it to me 😅 

But I believe that previously, we were not setting up and installing Terraform CLI before running our tests. The way the TF SDK works, it requires the CLI to be downloaded prior to running a test. Since we never set it up, the SDK was attempting to download the TF CLI for each test! 🤯 

Setting up the CLI prior to running tests fixes the issue of the TF SDK trying to download the CLI on every test. The flakiness was bound to happen for the amount of times we were downloading and reaching out to `releases.hashicorp.com` 😞